arduino/smartcar/smartcar.ino

```diff
@@ -1,116 +1,175 @@
-#include <Smartcar.h>
-
-const unsigned long PRINT_INTERVAL = 100;
-unsigned long previousPrintout     = 0;
-
-ArduinoRuntime arduinoRuntime;
-BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
-BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
-DifferentialControl control(leftMotor, rightMotor);
-
-GY50 gyroscope(arduinoRuntime, 37);
-
-const auto pulsesPerMeter = 600;
-const int TRIGGER_PIN           = 6; // D6
-const int ECHO_PIN              = 7; // D7
-const unsigned int MAX_DISTANCE = 400;
-
-const int NORMAL_SPEED = 40; // 30% of the motor capacity
-
-
-DirectionalOdometer leftOdometer{arduinoRuntime, smartcarlib::pins::v2::leftOdometerPins, []() { leftOdometer.update(); }, pulsesPerMeter};
-DirectionalOdometer rightOdometer{arduinoRuntime, smartcarlib::pins::v2::rightOdometerPins, []() { rightOdometer.update(); }, pulsesPerMeter};
-SmartCar car(arduinoRuntime, control, gyroscope, leftOdometer, rightOdometer);
-SR04 front(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
-
-void setup()
-{
-    Serial.begin(9600);
-    //car.enableCruiseControl();
-
-    car.setSpeed(NORMAL_SPEED); 
-
-}
-
-void loop()
-{
-   // Maintain the speed and update the heading
-    
-    car.setSpeed(30);
-    //car.update();
-
-   obstacleAvoidance();
-   //car.update();
-
-    
-}
-
-
-
-/**
- * ObstacleAvoidance: stops and rotate (calls rotateOnSpot)
- * 
-*/
-void obstacleAvoidance(){
-  
-  int distance = front.getDistance();
-  int rotationDegree = 90;
-    // Stop after moving 1 meter 
-    if (distance > 0 && distance< 200)
+    #include <MQTT.h>
+    #include <WiFi.h>
+    #ifdef __SMCE__
+    #include <OV767X.h>
```

Comment:
The sketch doesn't use Camera atm so no reason to include it

Reply:
Yes, forgot that one :)